### PR TITLE
fix: check VIDEO_STREAM_STATUS messages whether stream is running

### PR DESCRIPTION
**Pull request by RedFox20**
I discovered this bug when trying to add integration support for a multi-camera mavlink device with multiple streams.

The current code assumed that each Camera would always be streaming, but this is not always the case, and not guaranteed by MAVLink camera protocol. Especially with 3+ cameras, due to data rate limitations only 1 out 3 might be active.

QGC would be stuck assuming that the streams are running, but the camera system could be expecting for a VIDEO_START_STREAMING command for those particular streams.

The VIDEO_STREAM_STATUS `flags` field describes whether the Stream is actually running, and updating the `_currentStream` value according to those flags re-enables video stream switching.

I also slightly improved the camera debug logging to make it easier to debug multi-camera system issues.

### CODE REVIEW

File: src/Camera/QGCCameraControl.cc [hunk 1]
```diff
@@ -1600,45 +1601,53 @@ void
...
-                    _compID,                                // Target component
-                    MAV_CMD_VIDEO_START_STREAMING,          // Command id
-                    false,                                  // ShowError
-                    pInfo->streamID());                     // Stream ID
-                //-- Update stream status
-                _requestStreamStatus(static_cast<uint8_t>(pInfo->streamID()));
-            }
-            emit currentStreamChanged();
-            emit _vehicle->cameraManager()->streamChanged();
+    if (stream != _currentStream && 0 <= stream && stream < _streamLabels.count()) {
```
Comment (patrickelectric) [inline src/Camera/QGCCameraControl.cc]: you can just revert the logic, avoid the indentation level completely.

File: src/Camera/QGCCameraControl.cc [hunk 2]
```diff
@@ -1600,45 +1601,53 @@ void
...
+        if (pInfo->isRunning()) { // this is the current active stream
+            _currentStream = _streams.indexOf(pInfo);
+        } else { // we don't know which stream is active, check all of them
+            _currentStream = -1;
+            for (int i = 0; i < _streams.count(); i++) {
+                if (auto pStream = qobject_cast<QGCVideoStreamInfo*>(_streams[i]); pStream->isRunning()) {
+                    _currentStream = i;
+                    break;
+                }
+            }
```
Comment (patrickelectric) [inline src/Camera/QGCCameraControl.cc:1616]: you can use `std::find_if` and `std::distance`

**Review by patrickelectric — CHANGES_REQUESTED** [review]

**Comment by booo** [comment]
@RedFox20 can you update the PR with the suggested changes?

**Comment by booo** [comment]
@RedFox20 please reopen if you find the time to integrate the suggested changes. We would love to integrate your updated PR.

**Pull request closed without merging**